### PR TITLE
 #170947653 Docker file that will be used to create Docker image

### PR DESCRIPTION
**Pull request by EugeneMunya**
#### What does this PR do?
- This PR is adding Dockerfile for creating docker image
#### Description of Task to be completed?
- writing all commands that Docker will use to generate both docker image and docker container
#### How should this be manually tested?
- [clone the repo](https://github.com/andela/knights-bn-backend.git)
- run `npm i` to install the dependencies 
- see if your docker is installed by writing `docker version` if not install it [link](https://docs.docker.com/install/)
- run `docker build -t knights-bn-backend .` to create docker image.
- run `docker images ` to view your image
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#170947653](https://www.pivotaltracker.com/story/show/170947653)
#### Screenshots (if appropriate)
Screen Shot 2020-02-05 at 16 27 13
#### Questions:
N/A

### CODE REVIEW

**Review by Kagorora — APPROVED** [review]
LGTM

**Review by T2Wil — CHANGES_REQUESTED** [review]
Hey @EugeneMunya ,
What's going to happen after I run `docker build -t docker image name .`?

add the answer in the PR description, please.

**Comment by EugeneMunya** [comment]
> Hey @EugeneMunya ,
> What's going to happen after I run `docker build -t docker image name .`?
> 
> add the answer in the PR description, please.

Is 

> Hey @EugeneMunya ,
> What's going to happen after I run `docker build -t docker image name .`?
> 
> add the answer in the PR description, please.

Yes, just run that code and see if your image was created by running `docker images`

**Review by T2Wil — APPROVED** [review]

File: dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM node:10
+WORKDIR /knights
+COPY package.json ./
+RUN npm install
+COPY . .
+EXPOSE 4000
```
Comment (MuhanguziDavid) [inline dockerfile:6]: The port specified here is 4000 yet the app is now running on port 5000 as a default. Please change the default app port to 4000.

Comment (EugeneMunya) [reply dockerfile:6]: I have changed it

**Review by Niyonsengaeric — APPROVED** [review]

**Review by MuhanguziDavid — APPROVED** [review]
LGTM

**Pull request merged**